src/components/Checkout/BasketVerification.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+	<!-- NB: Reusing .order-totals .forced-width styling. -->
+	<div class="basket-verification order-totals">
+		<div
+			v-if="verificationState === 'verified'"
+			class="verification-verified forced-width"
+		>
+			<div class="forced-width">
+				Basket Verified!
+			</div>
+		</div>
+
+		<kv-lightbox
+			:visible="verificationState === 'required'"
+			class="verification-required"
+			title="Verify your email"
+		>
+			<p>To ensure your safety, we added an extra layer of security.</p>
+			<p>Once we verify your account, you can continue checking out!</p>
+			<kv-button @click.native="send">
+				Send verification link
+			</kv-button>
+		</kv-lightbox>
+
+		<kv-lightbox
+			:visible="verificationState === 'pending'"
+			class="verification-pending"
+			title="Verification link sent"
+		>
+			<p>We sent a validation link <span v-if="email" class="email">to {{ email }}</span>.</p>
+			<p>After receiving the email, follow the link provided to continue checking out with your Kiva Credit.</p>
+			<kv-button @click.native="send">
+				Resend email
+			</kv-button>
+			<p>
+				<router-link to="/help/contact-us">
+					Need help? Contact us
+				</router-link>
+			</p>
+		</kv-lightbox>
+	</div>
+</template>
+
+<script>
+import _get from 'lodash/get';
+import KvButton from '@/components/Kv/KvButton';
+import KvLightbox from '@/components/Kv/KvLightbox';
+import getVerificationState from '@/graphql/query/checkout/basketVerificationState.graphql';
+import startBasketVerificationMutation from '@/graphql/mutation/checkout/basketVerification.graphql';
+import logReadQueryError from '@/util/logReadQueryError';
+
+export default {
+	components: {
+		KvButton,
+		KvLightbox,
+	},
+	inject: ['apollo'],
+	data() {
+		return {
+			email: '',
+			verificationState: '',
+		};
+	},
+	computed: {
+		verificationRequired() {
+			return this.verificationState === 'required' || this.verificationState === 'pending';
+		},
+	},
+	created() {
+		// Create an observer for changes to the basket verification states
+		this.apollo.watchQuery({ query: getVerificationState })
+			.subscribe({
+				next: ({ data }) => {
+					const state = _get(data, 'shop.basketVerificationState', this.verificationState);
+					this.setState(state);
+				},
+				error: logReadQueryError,
+			});
+	},
+	methods: {
+		setState(state) {
+			this.verificationState = state;
+			this.$emit('verification-state', this.verificationState);
+			this.$emit('verification-required', this.verificationRequired);
```

Comment:
We usually have parent components feed state to child components rather than the other way around. I'm curious how this will work with the server-side render, because that's not clear to me. I think this should be ok, but I want to test some things out with it.

Reply:
I'm guessing this will just fail silently on the server, or maybe work intermittently, depending on timing of other operations... It's not promised based and has no mechanism to force the server renderer to wait on it.

I don't think there is a need to force that as it should run in the client as well...